crates/re_viewport/src/space_view_heuristics.rs

```diff
@@ -435,34 +434,18 @@ pub fn is_entity_processed_by_class(
     class: SpaceViewClassIdentifier,
     ent_path: &EntityPath,
     heuristic_ctx: HeuristicFilterContext,
-    query: &LatestAtQuery,
 ) -> bool {
-    let parts = ctx.space_view_class_registry.new_part_collection(class);
-    is_entity_processed_by_part_collection(
-        ctx.store_db.store(),
-        &parts,
-        ent_path,
-        heuristic_ctx.with_class(class),
-        query,
-    )
-}
+    let parts = &ctx.space_view_class_registry.new_part_collection(class);
+    let heuristic_ctx = heuristic_ctx.with_class(class);
 
-/// Returns true if an entity is processed by any of the given [`re_viewer_context::ViewPartSystem`]s.
-fn is_entity_processed_by_part_collection(
-    store: &re_arrow_store::DataStore,
-    parts: &ViewPartCollection,
-    ent_path: &EntityPath,
-    ctx: HeuristicFilterContext,
-    query: &LatestAtQuery,
-) -> bool {
-    let timeline = Timeline::log_time();
-    let components = store
-        .all_components(&timeline, ent_path)
-        .unwrap_or_default()
-        .into_iter()
-        .collect();
-    for part in parts.iter() {
-        if part.heuristic_filter(store, ent_path, ctx, query, &components) {
+    let empty_entity_set = IntSet::default();
+    for (id, visualizer) in parts.iter_with_identifiers() {
+        let entites_with_matching_indicator = ctx
+            .entities_with_matching_indicator_per_visualizer
+            .get(&id)
+            .unwrap_or(&empty_entity_set);
```

Comment:
If this is an empty set, do we still need to run: `visualizer.heuristic_filter`, or can we just return false directly?

Reply:
Hmm practically yes. But technically `heuristic_filter` decides whehter to use `entites_with_matching_indicator` in the first place 🤔
I'm a bit undecided. leaving it for now